src/test/java/com/ning/http/client/async/grizzly/GrizzlySimpleAsyncHttpClientAllowingUTF8Test.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2012 Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.ning.http.client.async.grizzly;
+
+import static com.ning.http.client.multipart.PartBase.GRIZZLY_ALLOW_UTF8;
+
+import org.testng.annotations.BeforeClass;
+
+import com.ning.http.client.AsyncHttpClient;
+import com.ning.http.client.AsyncHttpClientConfig;
+import com.ning.http.client.async.ProviderUtil;
+import com.ning.http.client.async.SimpleAsyncHttpClientTest;
+import com.ning.http.client.providers.grizzly.GrizzlyAsyncHttpProvider;
+
+public class GrizzlySimpleAsyncHttpClientAllowingUTF8Test extends SimpleAsyncHttpClientTest {
+
+    @BeforeClass
+    public void beforeClass() {
+      System.setProperty(GRIZZLY_ALLOW_UTF8, "true");
```

Comment:
As the property is static won't this mean the first load sticks?

Reply:
Yes, it performs every test in SimpleAsyncHttpClientTest obtaining the string for each header as UTF-8. GrizzlySimpleAsyncHttpClient performs the same tests without the system property. GrizzlySimpleAsyncHttpClientAllowingUTF8Test also runs the tests where every character is ASCII valid.

Reply:
Ignore previous comment. Changed to use Reflection.